src/Orchard/Environment/Configuration/ShellSettings.cs

```diff
@@ -15,6 +15,15 @@ public class ShellSettings {
         private string[] _modules;
         private readonly IDictionary<string, string> _values;
 
+        public string[] _globalPathPrefixes = { };
+        public string[] GlobalPathPrefixes { get { return _globalPathPrefixes; } }
+        private static string[] defaultGlobalPrefixes = { "~/Media" }; 
+
+        public void updateGlobalPathPrefixes(IEnumerable<string> locations)
+        {
+            _globalPathPrefixes = defaultGlobalPrefixes.Concat(locations).ToArray();
+        }
+

```

Comment:
All this seems off. Why in ShellSettings if it's not actually part of shell settings (nor is it persisted)? Names are unconventional too.